keyboards/hfdkb/keyboard_sw/mk47/mk47.c

```diff
@@ -0,0 +1,372 @@
+            } else {
+                Lkey_flag = 0;
+            }
+            return true;
+        case KC_SPC:
+            if (fn_make_flag && record->event.pressed) {
+                if (while_test_flag) {
+                    while_test_flag = false;
+                    rgb_matrix_init();
+                } else {
+                    Lkey_flag    = true;
+                    current_time = timer_read();
+                    scancode     = KC_SPC;
+                }
+                return false;   
+            } else {
+                Lkey_flag = 0;
+            }
+            return true;
+        case RGB_VAI:
+            if ((fn_make_flag && record->event.pressed)) {
+                if ((RGB_HSV_level = (uint8_t)rgb_matrix_get_val() / (RGB_MATRIX_MAXIMUM_BRIGHTNESS / 4)) < 4) {
+                    RGB_HSV_level++;
+                    rgb_matrix_config.hsv.v = (uint8_t)(RGB_MATRIX_MAXIMUM_BRIGHTNESS / 4) * RGB_HSV_level;
+                }
+                rgb_hsv_updata_user();
+            }
+            return false;
+        case RGB_VAD:
+            if ((fn_make_flag && record->event.pressed)) {
+                if ((RGB_HSV_level = (uint8_t)rgb_matrix_get_val() / (RGB_MATRIX_MAXIMUM_BRIGHTNESS / 4)) > 0) {
+                    RGB_HSV_level--;
+                    rgb_matrix_config.hsv.v = (uint8_t)(RGB_MATRIX_MAXIMUM_BRIGHTNESS / 4) * RGB_HSV_level;
+                }
+                rgb_hsv_updata_user();
+            }
+            return false;
+        case RGB_HUI:
+            if ((fn_make_flag && record->event.pressed)) {
+                if ((RGB_HSV_level = (uint8_t)rgb_matrix_get_hue() / (UINT8_MAX / 9)) < 9) {
+                    RGB_HSV_level++;
+                }else{
+                    RGB_HSV_level = 1;
+                }
+                rgb_matrix_config.hsv.h = (uint8_t)(UINT8_MAX / 9) * RGB_HSV_level;
+                rgb_hsv_updata_user();
+            }
+            return false;
+
+        case RGB_SPI:
+            if ((fn_make_flag && record->event.pressed)) {
+                if ((RGB_HSV_level = (uint8_t)rgb_matrix_get_speed() / (UINT8_MAX / 5)) < 5) {
+                    RGB_HSV_level++;
+                    rgb_matrix_set_speed((uint8_t)(UINT8_MAX / 5) * RGB_HSV_level);
+                }
+            }
+            return false;
+        case RGB_SPD:
+            if ((fn_make_flag && record->event.pressed)) {
+                if ((RGB_HSV_level = (uint8_t)rgb_matrix_get_speed() / (UINT8_MAX / 5)) > 0) {
+                    RGB_HSV_level--;
+                    rgb_matrix_set_speed((uint8_t)(UINT8_MAX / 5) * RGB_HSV_level);
+                }
+            }
+            return false;     
+#ifdef VIA_ENABLE
+        case QK_KB: {
+            if (record->event.pressed) {
+                via_eeprom_set_valid(false);
+                eeconfig_init_via();
+            }
+            return false;
+        }
+#endif
+     default:
+            return process_record_user(keycode, record);
```

Comment:
`process_record_user` is already being checked, above. No need to repeat, or you may get double events. 
```suggestion
            return true;
```